### PR TITLE
Tasks/blog 116 teracy dev kubernetes

### PR DESCRIPTION
**Pull request by hoatle**
need @phuonglm's review for technical content
need @hoavt to review, approve and merge when it's good enough

to review:

```
$ export DOCKER_IMAGE_REVIEW=hoatle/teracy-blog:tasks-BLOG-116-teracy-dev-kubernetes && \
  docker-compose pull review && \
  docker-compose up review
```

### CODE REVIEW

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+Working with Kubernetes requires `kubectl` client to be available, and if you're starting to use
+GKE (Google Container Engine), `gcloud` (google cloud sdk) client should be available, too.
+
+So let's find a way to extend teracy-dev to install `kubectl` and `gcloud`.
+
+
+## Extending teracy-dev
+
+You can extend `teracy-dev`'s VM by your own choice of operating system and automate the provision
+process by your own choice of configuration software. There is only one limitation is your imagination :-).
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: You can extend `teracy-dev`'s VM by your own choice of operating system and automate the provision
process by your own choice of configuration software. --> You can extend....and automate the provision process by selecting your desired configuration software. 


There is only one limitation is your imagination :-). -->There is only one limitation that is your imagination :-).

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+to use `Ansible` for some future projects, too.
+
+
+## Installing ChefDK
+
+We're going to use `Acme 101` project which is used to show the best practices from `teracy-dev` applied
+for organizations.
+
+To work with Chef cookbooks, we need to install ChefDK. Fortunately, there is already an available cookbook
+for us to use to install ChefDK automatically on to our VM: https://supermarket.chef.io/cookbooks/chef-dk
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: on to our VM --> on our VM:

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+
+Usually, we have `dev-setup` directory to extend `teracy-dev` (acme-dev in this case). The initial
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: acme-dev in this case -->`acme-dev` in this case

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+chef_dk 'my_chef_dk' do
+  global_shell_init true
+  action :install
+end
+```
+
+Make sure you have `berks-cookbooks` paths that `vagrant` can look up. The configuration step should
+be like this: https://github.com/acme101/kubernetes-dev-setup/tree/2-configuration
+
+After that, `$ vagrant reload --provision` and voila, you should have ChefDk installed.
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: have ChefDk installed. --> have `ChefDk` installed.

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+That's how we use Chef cookbooks to manage the VM's software automatically. We can do the same with
+all other types of Chef cookbooks shared and opensourced from the public Chef Supermarket: https://supermarket.chef.io/
+You can use all the public shared cookbooks to do almost anything you want for your VM.
+
+However, sometimes, there is not available cookbook that we want, then it's time we should
+build new cookbooks from scratch.
+
+## Creating new Chef cookbooks
+
+From the step above, we have `ChefDK` available to work with Chef cookbooks. To learn how to use it,
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: From the steps above, we...

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+       Deleting kitchen sandbox at /home/vagrant/.dokken/kitchen_sandbox/1000b8c847-smoke-gcloud-ubuntu-1604
+       Deleting verifier sandbox at /home/vagrant/.dokken/verifier_sandbox/1000b8c847-smoke-gcloud-ubuntu-1604
+       Finished destroying <smoke-gcloud-ubuntu-1604> (0m10.67s).
+       Finished testing <smoke-gcloud-ubuntu-1604> (1m55.61s).
+-----> Kitchen is finished. (4m12.76s)
+```
+
+That's how we develop and test the cookbook on local dev.
+
+You can see the cookbook here at: https://github.com/teracyhq-incubator/kubernetes-stack-cookbook
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: You can see the cookbook [here](https://github.com/teracyhq-incubator/kubernetes-stack-cookbook).

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+for organizations.
+
+To work with Chef cookbooks, we need to install ChefDK. Fortunately, there is already an available cookbook
+for us to use to install ChefDK automatically on to our VM: https://supermarket.chef.io/cookbooks/chef-dk
+
+
+Usually, we have `dev-setup` directory to extend `teracy-dev` (acme-dev in this case). The initial
+`dev-setup` content should be like this: https://github.com/acme101/kubernetes-dev-setup/tree/0-initial
+
+To install ChefDK, we must install the chef-dk cookbook and use it as I did as follows:
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: To install ChefDK, we must install the chef-dk cookbook  --> To install `ChefDK`, we must install the `chef-dk` cookbook

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+
+- Install vendor cookbooks with the following commands within the VM:
+
+```
+$ ws
+$ cd dev-setup/chef
+$ docker-compose up
+```
+
+After that, `$ vagrant rsync-back` to sync back the changes from the VM to the host machine and the
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: - Sync back the changes from the VM to the host machine:
``
$ vagrant rsync-back
```
The updated content should be....

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+updated content should be like this: https://github.com/acme101/kubernetes-dev-setup/tree/1-dependency
+
+
+Now, to install chef-dk, just add the following Ruby code to `default.rb` recipe, it's never so easy:
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: - Install `chef-dedk` by adding the following Ruby code to `default.rb` recipe, it's never so easy:

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown (hunk 4 above)
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: - After that,....

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown [hunk 10]
```diff
@@ -0,0 +1,333 @@
+---
+layout: post
+title: "How to extend teracy-dev to work with Kubernetes"
+author:
```
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: lack of author name

**Comment by hoatle** [comment]
@hoavt PR updated.

File: source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown (hunk 1 above)
Comment (hoavt) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: this is not applied yet

Comment (hoatle) [inline source/_posts/2017-06-15-how-to-extend-teracy-dev-to-work-with-kubernetes.markdown]: "You can extend....and automate the provision process by selecting your desired configuration software" => I'll keep the current version.

There is only one limitation is your imagination :-). --> "The only limitation is your imagination" updated.

**Pull request merged**